Refactor Login Backend

braunnymadrigal opened the pull request:
[JIRA Task](https://infinipay.atlassian.net/jira/software/projects/SCRUM/boards/1/backlog?selectedIssue=SCRUM-89)

braunnymadrigal committed:
creadas las clases abstractas GeneralController y GeneralRepository

braunnymadrigal committed:
refactor listo

ahora login esta dividido en LoginController, LoginRepository, ILogin, UserModel, LoginUserModel y Login(business rules)

24danielcr commented on back-end/back-end/API/LoginController.cs:
```diff
@@ -1,63 +1,38 @@
         public IActionResult Login([FromBody] LoginUserModel loginUserModel)
         {
-            IActionResult iActionResult = BadRequest("UNKNOWN ERROR");
+            IActionResult iActionResult = BadRequest("Uknown error.");
```
Pequeño error de gramática. Uknown -> Unknown.

24danielcr commented on back-end/back-end/Infraestructure/LoginRepository.cs:
```diff
@@ -1,238 +1,153 @@
+            var password = (byte[])(dataRow["usuarioContrasena"]);
+            var numAttempts = Convert.ToString(dataRow["usuarioNumIntentos"]);
+            var lastBlock = Convert.ToString(dataRow["usuarioFechaExactaBloqueo"]);
+            var empleadoId = Convert.ToString(dataRow["empleadoId"]);
```
Nombre en español, igual en empleadorId, pasar a employeeId y employerId.

24danielcr commented on back-end/back-end/Infraestructure/LoginRepository.cs:
```diff
@@ -1,238 +1,153 @@
-                }
-            }
-            return userModel;
+        private UserModel TransformDataTableIntoUserModel(DataTable dataTable)
```
Función sobrepasa las 20 lineas, crear otra función para chequeo de errores, y buscar otra forma de revisar nulos ? Si se fuera a agregar otro elemento, se debe agregar otro if.

24danielcr reviewed (changes requested):
Revisar comentarios, en general buen refactor, sólo unos pequeños detalles por resolver.

braunnymadrigal committed:
creado el abstract connection repository

con este se hacen todas las conecciones a la base de datos

braunnymadrigal committed:
correccion de visibilidad en el abstract connection repository

braunnymadrigal committed:
creada la implementacion concreta del abstract connection repository

braunnymadrigal committed:
creada una interfaz para el LoginRepository

braunnymadrigal committed:
creado un UtilityRepository con su respectiva interfaz

este va a resultar extremadamente util para definir metodos usamos de forma muy frecuente a la hora de convertir tablas a modelos

braunnymadrigal committed:
loginRepository completamente corregido en cuanto a clean code

braunnymadrigal committed:
correcciones de estilo de la clase de negocio Login

braunnymadrigal committed:
correcion de ortografia en un exception del LoginController

braunnymadrigal committed:
ahora los LoginController y Login cumplen el Dependency Inversion Principle

braunnymadrigal commented:
Listo, acabo de corregir todo lo que me dijo. Gracias por la revisión.
